spotify_player/src/event/mod.rs

```diff
@@ -346,6 +383,40 @@ fn handle_show_actions_on_artist(
     }
 }
 
+/// Handle a global action, currently this is only used to target
+/// the currently playing song instead of the selection.
+fn handle_global_action(
+    action: Action,
+    target: ActionTarget,
+    client_pub: &flume::Sender<ClientRequest>,
+    state: &SharedState,
+    ui: &mut UIStateGuard,
+) -> Result<bool> {
+    if target == ActionTarget::PlayingTrack {
+        let player = state.player.read();
+        let data = state.data.read();
+
+        let currently_playing = match player.current_playing_track() {
+            Some(track) => track,
+            None => anyhow::bail!("No currently playing track"),
+        };
+        let track = match Track::try_from_full_track(currently_playing.clone()) {
+            Some(track) => track,
+            None => anyhow::bail!("No currently playing track"),
+        };
```

Comment:
normally, we don't return error when there is nothing to act on when handling a key event, e.g. no playing track. See how `Command::ShowActionsOnCurrentTrack` is handled in `event/mod.rs`:

```rust
  if let Some(track) = state.player.read().current_playing_track() {
      if let Some(track) = Track::try_from_full_track(track.clone()) {
          let data = state.data.read();
          ...
```